sstables/compaction.cc

```diff
@@ -692,9 +692,13 @@ class regular_compaction : public compaction {
     sstable_set _set;
     // used to incrementally calculate max purgeable timestamp, as we iterate through decorated keys.
     std::optional<sstable_set::incremental_selector> _selector;
-    // sstable being currently written.
-    shared_sstable _sst;
-    std::optional<sstable_writer> _writer;
+    // sstables being currently written.
+    struct sst_writer {
+        shared_sstable sst;
+        std::optional<sstable_writer> writer;
```

Comment:
looks like writer no longer needs to be optional.

Reply:
You are right.

I am trying to minimize changes and was a bit afraid of how this would interfere with LCS,
but I should be able to add a patch on top oto change the interface.